script/install.sh

```diff
@@ -105,45 +114,67 @@ linux() {
         ;;
     esac
 
-    # Unpack
     rm -rf "$HOME/.local/zed$suffix.app"
-    mkdir -p "$HOME/.local/zed$suffix.app"
-    tar -xzf "$temp/zed-linux-$arch.tar.gz" -C "$HOME/.local/"
 
-    # Setup ~/.local directories
-    mkdir -p "$HOME/.local/bin" "$HOME/.local/share/applications"
+    [ "$operation" = install ] && {
+        # Unpack
+        mkdir -p "$HOME/.local/zed$suffix.app" && tar -xzf "$temp/zed-linux-$arch.tar.gz" -C "$HOME/.local/"
 
-    # Link the binary
-    if [ -f "$HOME/.local/zed$suffix.app/bin/zed" ]; then
-        ln -sf "$HOME/.local/zed$suffix.app/bin/zed" "$HOME/.local/bin/zed"
-    else
-        # support for versions before 0.139.x.
-        ln -sf "$HOME/.local/zed$suffix.app/bin/cli" "$HOME/.local/bin/zed"
-    fi
+        # Setup ~/.local directories
+        mkdir -p "$HOME/.local/bin" "$HOME/.local/share/applications"
+
+        # Link the binary
+        if [ -f "$HOME/.local/zed$suffix.app/bin/zed" ]; then
+            ln -sf "$HOME/.local/zed$suffix.app/bin/zed" "$HOME/.local/bin/zed"
+        else
+            # support for versions before 0.139.x.
+            ln -sf "$HOME/.local/zed$suffix.app/bin/cli" "$HOME/.local/bin/zed"
+        fi
+    }
+
+    [ "$operation" = uninstall ] && rm -f "$HOME/.local/bin/zed"
 
     # Copy .desktop file
     desktop_file_path="$HOME/.local/share/applications/${appid}.desktop"
-    cp "$HOME/.local/zed$suffix.app/share/applications/zed$suffix.desktop" "${desktop_file_path}"
-    sed -i "s|Icon=zed|Icon=$HOME/.local/zed$suffix.app/share/icons/hicolor/512x512/apps/zed.png|g" "${desktop_file_path}"
-    sed -i "s|Exec=zed|Exec=$HOME/.local/zed$suffix.app/libexec/zed-editor|g" "${desktop_file_path}"
+
+    [ "$operation" = install ] && {
+        cp "$HOME/.local/zed$suffix.app/share/applications/zed$suffix.desktop" "${desktop_file_path}"
+        sed -i "s|Icon=zed|Icon=$HOME/.local/zed$suffix.app/share/icons/hicolor/512x512/apps/zed.png|g" "${desktop_file_path}"
+        sed -i "s|Exec=zed|Exec=$HOME/.local/zed$suffix.app/libexec/zed-editor|g" "${desktop_file_path}"
+    }
+
+    [ "$operation" = uninstall ] && rm -f "$desktop_file_path"
+
+    return 0
 }
 
 macos() {
     echo "Downloading Zed"
     curl "https://zed.dev/api/releases/$channel/latest/Zed-$arch.dmg" > "$temp/Zed-$arch.dmg"
```

Comment:
I don't think we need to download a `.dmg` when we uninstall, right? We also don't need to mount it etc. So I think a larger `if/else` makes more sense

Reply:
Yeah, we don't need to. But `.app` file name is being inferred from the `dmg` file downloaded.

If it's always going to be `Zed.app` it can be hard-coded.